### PR TITLE
#1036 added auto-vacation for negative awards

### PR DESCRIPTION
**Pull request by krzyk**
#1036 
- added implementation for automatic vacation for 8 negative awards in last 16 days

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1154](https://github.com/zerocracy/farm/issues/1154) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1154](https://github.com/zerocracy/farm/issues/1154) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1652147943); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy [hunk 1]
```diff
@@ -17,15 +17,34 @@
...
 
 def exec(Project project, XML xml) {
   new Assume(project, xml).type('Award points were added')
   ClaimIn claim = new ClaimIn(xml)
-  Integer points = Integer.parseInt(claim.param('points'))
-  if (points >= 0) {
-    return
+
+  String login = claim.param('login')
+
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy]: @krzyk remove this newline pls

File: src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy [hunk 2]
```diff
@@ -17,15 +17,34 @@
...
+  Farm farm = binding.variables.farm
+  People people = new People(farm).bootstrap()
+  if (!people.vacation(login)) {
+    def policy = new Policy()
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy]: @krzyk does this need to be ``def``? I never saw ``def`` until now, I think it should be the concrete type, ``Policy``

File: src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy [hunk 3]
```diff
@@ -17,15 +17,34 @@
...
+    if (
+      new Filtered<>(
+        { int points -> (points < 0) },
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy]: @krzyk is ``int`` needed here?

File: src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy [hunk 4]
```diff
@@ -17,15 +17,34 @@
...
+import org.cactoos.collection.Filtered
 
 def exec(Project project, XML xml) {
   new Assume(project, xml).type('Award points were added')
   ClaimIn claim = new ClaimIn(xml)
-  Integer points = Integer.parseInt(claim.param('points'))
-  if (points >= 0) {
-    return
+
+  String login = claim.param('login')
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy]: @krzyk I would rename this variable to ``user`` -- the user in question, the one which we are checking.

File: src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy [hunk 5]
```diff
@@ -17,15 +17,34 @@
...
+import com.zerocracy.Policy
 import com.zerocracy.Project
 import com.zerocracy.farm.Assume
 import com.zerocracy.pm.ClaimIn
+import com.zerocracy.pmo.Awards
+import com.zerocracy.pmo.People
+import org.cactoos.collection.Filtered
 
 def exec(Project project, XML xml) {
   new Assume(project, xml).type('Award points were added')
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy]: @krzyk Shouldn't it also assume that the project is not PMO? I saw that assumption in all stackholders until now

File: src/main/java/com/zerocracy/pmo/Awards.java [hunk 6]
```diff
@@ -170,6 +173,34 @@ public int total() throws IOException {
...
+     * @return List of awards
+     * @throws IOException If fails
+     */
+    public List<Integer> awards(final int days) throws IOException {
+        try (final Item item = this.item()) {
+            return new Mapped<>(
+                Integer::parseInt,
+                new Xocument(item.path()).xpath(
+                    new JoinedText(
+                        "",
```
Comment (amihaiemil) [inline src/main/java/com/zerocracy/pmo/Awards.java]: @krzyk Why are you joining empty string here?

**Comment by krzyk** [comment]
@amihaiemil fixed, please check

File: src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy (hunk 3 above)
Comment (amihaiemil) [reply src/main/groovy/com/zerocracy/stk/pmo/awards/check_automatic_vacation.groovy]: @krzyk what about this? I don't think ``int`` is needed here. Or does checkstyle complain if it's missing?

**Comment by amihaiemil** [comment]
@krzyk hmmm, now Travis fails, and it seems to be the test for this stackholder:

```
Caused by: java.lang.AssertionError: Vacation mode is "off"
Expected: is <true>
     but: was <false>
```

**Comment by krzyk** [comment]
@amihaiemil the error was because `JoinedText` first parameter `""` is the joining string, so when I removed it it was joining using `/awards/award ...` :)

**Comment by amihaiemil** [comment]
@krzyk I see; Just one unanswered question: 

```
{ int points -> (points < 0) },

```

I think the ``int`` is not needed there, or?

**Comment by krzyk** [comment]
@amihaiemil fixed

**Comment by amihaiemil** [comment]
@krzyk nice, thanks

**Comment by amihaiemil** [comment]
@g4s8 good to merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1154](https://github.com/zerocracy/farm/issues/1154) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**